.github/workflows/ci.yml

```diff
@@ -368,17 +338,18 @@ jobs:
   test-aws-lambda:
     name: Test AWS Lambda
     runs-on: ubuntu-latest
-    needs: [ docker-build-test ]
+    needs: [ build ]
     steps:
       - name: Checkout sources
         uses: actions/checkout@v4
-      - name: Download build artifact cross-build
+      - name: Download build artifact build-x86_64-unknown-linux-gnu
         uses: actions/download-artifact@v4
         with:
-          name: cross-build
+          name: build-x86_64-unknown-linux-gnu
+          path: target_release/x86_64-unknown-linux-gnu
       - run: tests/test-aws-lambda.sh
         env:
-          MARTIN_BIN: x86_64-unknown-linux-musl/martin
+          MARTIN_BIN: target_release/x86_64-unknown-linux-gnu/martin
```

Comment:
I now get why this is done this way. AWS does seem to like their musl build.

@nyurik
could you help me with why this is a different pipeline step?
I have a bit of trouble to understand why this is not just in the `docker-build-test`-step.
All dependencys already exist there => why upload and download here?

(don't look at the rest of the code, it is a bit WIP)